### ISSUE TITLE
Integration test flake: nil pointer error in federation TestNamespaceController

### ISSUE THREAD

**Issue by ericchiang** [kubernetes/kubernetes#30788]
Got a nil pointer error when testing #30392

cc @mwielgus (based on git history)

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/30392/kubernetes-pull-test-unit-integration/40033/

```
<autogenerated>:31: 
                        
	Error Trace:	namespace_controller_test.go:86
	
	Error:		Expected value not to be nil.
	
panic: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:481 +0x3ff
/usr/local/go/src/testing/testing.go:467 +0x22a
/usr/local/go/src/runtime/panic.go:443 +0x521
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/namespace/namespace_controller_test.go:87 +0x1a48
/usr/local/go/src/testing/testing.go:473 +0xdd
/usr/local/go/src/testing/testing.go:582 +0xae3from junit_v1,apps-v1alpha1,authentication.k8s.io-v1beta1,authorization.k8s.io-v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.k8s.io-v1alpha1_20160817-100713.xml
```

**Comment by wojtek-t** [comment]
@mwielgus - is this fixed?

**Comment by krousey** [comment]
@mwielgus  This has been marked with the "needs attention" label and needs to be addressed for the 1.4 release. Could you please

- close if it's fixed
- update on your progress to fix it
- make a case for it not to block the release
- or find a new owner who can do one of the above soon.

Once you or the new owner update this issue and are actively working on it, you can remove the "needs attention" label.

**Comment by mwielgus** [comment]
Should be fixed.
